### PR TITLE
chore: add a build script for scenarios

### LINKED ISSUE #3930: Streamline process for adding scenarios to Component explorer site
# Abstract

The creation of component scenarios for the component explorer site is unintuitive and should be made easier for contributors.

## Requirements

Ideally instead of writing a `DataDictionary` which is inherently difficult, the scenarios should instead be individual `.html` files located in a scenarios folder. This way a node script can be run to scrape them for content and inject them into the site.

- Create a Node.js script to read contents from a `scenarios` folder in `@microsoft/fast-components` package for each component
- Derive the scenario name from the file name
- Run this Node.js script during build of the component explorer
- Convert the internals of the Node.js script to a `DataDictionary`

### PR DESCRIPTION
**Pull request by janechu**
# Description

This adds a build step to convert the newly added scenario `html` files into javascript files with exports that can be read by the component explorer site. The html from these files be converted in the component explorer using a data mapping utility from `@microsoft/fast-tooling` into the `DataDictionary` format after they are imported to the site. The reason this can't be done in the build script is due to the fact that the `@microsoft/fast-tooling` project only exports in the ES6 format.

## Motivation & context

Continues the work on #3930 

## Issue type checklist

- [x] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

**Comment by nicholasrice** [comment]
@janechu does this affect distributed NPM packages? If not I think it should be a chore.

File: sites/fast-component-explorer/build/convert-scenarios.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                const scenarios = [];
+
+                if (!fs.existsSync(path.resolve(exportPath, componentFolderName))) {
+                    fs.mkdirSync(path.resolve(exportPath, componentFolderName));
+                }
+
+                templates.forEach(template => {
+                    scenarios.push({
+                        displayName: template.getAttribute("title"),
+                        html: template.innerHTML,
```
Comment (nicholasrice) [inline sites/fast-component-explorer/build/convert-scenarios.js:54]: Probably not a big deal here, but FYI `<template>` elements have a special property for accessing the content and stamping it out: https://developer.mozilla.org/en-US/docs/Web/API/HTMLTemplateElement/content

**Review by nicholasrice — APPROVED** [review]

**Comment by janechu** [comment]
@nicholasrice so, this is a feature for the site technically and it's a private package. It seems like a nice thing to have in the changelog even if we aren't distributing it. I would have it be a "chore" if it affected one of our distributed packages.

File: sites/fast-component-explorer/build/convert-scenarios.js (hunk 1 above)
Comment (janechu) [reply sites/fast-component-explorer/build/convert-scenarios.js:54]: I would do that, but I actually need this to be a string of the internal HTML.

**Review by awentzel — APPROVED** [review]

**Pull request merged**